### ISSUE TITLE
ACE_Log_Msg %m without error number

### ISSUE THREAD

**Issue by likema** [DOCGroup/ACE_TAO#862]
`ACE_Log_Msg` `%m` only prints error message of `errno`. If program runs in Spanish (`LANG=es-ES`) Linux, the error message of `errno` is Spanish that is hard to understand for persons who do not know Spanish.

As my practice, I wrote log message as 
practise
```cpp
ACE_DEBUG((LM_DEBUG,
           "..., errno=%d, %m\n",
           ..., ACE_OS::last_error()));
```

Should I add `errno= ...` content` into `ACE_Log_Msg` `%m` ?

Or

Should I add a new format specifier into `ACE_Log_Msg`, such as one of the following unused characer?

* H, h
* J, j
* K, k
* L
* O
* U
* V, v
* Y, y

**Comment by jwillemsen** [comment]
I think we should add errno in front of the %m message, what do you think @mitza-oci and @shuston

**Comment by shuston** [comment]
I think that, given the original problem statement (message is in spanish and not understandable), the real solution is to set the run-time language to something understandable.
But if this is a case where the developer is collecting logs from a user who speaks a different language, that's certainly a good rationale for adding the errno value.

Maybe something like:  "(22) error message text"

Alternately could add 'v' (verbose) and leave 'm' alone to avoid problems with current applications.

**Comment by jwillemsen** [comment]
I can't think of a reason why adding (errno) to the log message would break an application, as mentioned by the reporter it is a language dependent string which I think nobody could parse, just for developers to analyze in case of a failure

**Comment by shuston** [comment]
People do lots of things with log files so it's hard to say. However, if you all feel that adding (errno) to the %m message is the way to go I will support that.

**Comment by mitza-oci** [comment]
Adding this behavior as a different format specifier would avoid the problem of someone's existing log format that already inserts the numerical errno now containing a duplicate.

**Comment by jwillemsen** [comment]
Yes, the user could decide to log errno himself, so it looks better to me to just leave %m as is, the user can log errno himself

**Comment by likema** [comment]
I think adding a new different format specifier for this scenario is a good idea

**Comment by jwillemsen** [comment]
Just adding a new format specifier that combines two existing ones doesn't seem a good idea to me. Maybe extend the %m documentation a little bit to warn that this could be a message in the user language, not necessary English

**Comment by likema** [comment]
Maybe it is a good practice to me for logging as `errno=%d %m` for multilingual environment. 

But it also requires writing more code for every **syscall** related functions error log.
